src/wolfip.c

```diff
@@ -1828,23 +1910,51 @@ static void close_socket(struct tsocket *ts)
     memset(ts, 0, sizeof(struct tsocket));
 }
 
+static struct tsocket *wolfIP_socket_from_fd(struct wolfIP *s, int sockfd)
+{
+    if (!s || sockfd < 0)
+        return NULL;
+    if (sockfd & MARK_TCP_SOCKET) {
+        if ((sockfd & ~MARK_TCP_SOCKET) >= MAX_TCPSOCKETS)
+            return NULL;
+        return &s->tcpsockets[sockfd & ~MARK_TCP_SOCKET];
+    } else if (sockfd & MARK_UDP_SOCKET) {
+        if ((sockfd & ~MARK_UDP_SOCKET) >= MAX_UDPSOCKETS)
+            return NULL;
+        return &s->udpsockets[sockfd & ~MARK_UDP_SOCKET];
+    } else if (sockfd & MARK_ICMP_SOCKET) {
```

Comment:
tiny suggestion: `sockfd & MARK_ICMP_SOCKET` and `sockfd & ~MARK_ICMP_SOCKET` and related appear a lot. With a small typo one could be the other on accident.

Recommend replacing with two very different defines that are less susceptible to typos.